syntax = "proto3";

package spaceship;

enum Alignment {
    ALLY = 0;
    ENEMY = 1;
}

enum Class {
    CORVETTE = 0;
    FRIGATE = 1;
    CRUISER = 2;
    DESTROYER = 3;
    CARRIER = 4;
    DREADNOUGHT = 5;
}

message Officer {
    string first_name = 1;
    string last_name = 2;
    string rank = 3;
}

message Spaceship {
    Alignment alignment = 1;
    string name = 2;
    Class class_ = 3;
    float length = 4;
    int32 crew_size = 5;
    bool armed = 6;
    repeated Officer officers = 7;
}

message Coordinates {
    string coordinates = 1;
}

service ReportingService {
    rpc ReportSpaceships(Coordinates) returns (stream Spaceship);
}
